include/deal.II/matrix_free/cuda_fe_evaluation.h

```diff
@@ -38,6 +38,22 @@ DEAL_II_NAMESPACE_OPEN
  */
 namespace CUDAWrappers
 {
+  /**
+   * Compute the dof/quad index for a given thread id, dimension, and
+   * number of points in each space dimensions.
+   */
+  template <int dim, int n_points_1d>
+  __device__ inline unsigned int
+  compute_index()
```

Comment:
Thinking about this more, I think that we don't need this function to be public. Can you put it in `internal` namespace?